### PR TITLE
[v10.0.x] Alerting: Rewrite range to instant queries if possible

### PR DESCRIPTION
**Pull request by JohnnyQQQQ**
Backport 934ba1aaa1fa4c1b9f1dcc0d6753c67a23c4f466 from #69976

### CODE REVIEW

File: packages/grafana-data/src/types/featureToggles.gen.ts [hunk 1]
```diff
@@ -97,4 +97,13 @@ export interface FeatureToggles {
...
   enableDatagridEditing?: boolean;
+  dataSourcePageHeader?: boolean;
+  extraThemes?: boolean;
+  lokiPredefinedOperations?: boolean;
+  pluginsFrontendSandbox?: boolean;
+  sqlDatasourceDatabaseSelection?: boolean;
+  cloudWatchLogsMonacoEditor?: boolean;
+  exploreScrollableLogsContainer?: boolean;
+  recordedQueriesMulti?: boolean;
+  alertingLokiRangeToInstant?: boolean;
```
Comment (JohnnyQQQQ) [inline packages/grafana-data/src/types/featureToggles.gen.ts:108]:
```suggestion
  alertingLokiRangeToInstant?: boolean;
```

File: docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md [hunk 2]
```diff
@@ -111,6 +111,13 @@ Alpha features might be changed or removed without prior notice.
...
+| `extraThemes`                      | Enables extra themes                                                                                         |
+| `lokiPredefinedOperations`         | Adds predefined query operations to Loki query editor                                                        |
+| `pluginsFrontendSandbox`           | Enables the plugins frontend sandbox                                                                         |
+| `cloudWatchLogsMonacoEditor`       | Enables the Monaco editor for CloudWatch Logs queries                                                        |
+| `exploreScrollableLogsContainer`   | Improves the scrolling behavior of logs in Explore                                                           |
+| `recordedQueriesMulti`             | Enables writing multiple items from a single query within Recorded Queries                                   |
+| `alertingLokiRangeToInstant`       | Rewrites eligible loki range queries to instant queries                                                      |
```
Comment (JohnnyQQQQ) [inline docs/sources/setup-grafana/configure-grafana/feature-toggles/index.md:120]:
```suggestion
| `alertingLokiRangeToInstant`       | Rewrites eligible loki range queries to instant queries                                                      |
```

File: pkg/services/featuremgmt/registry.go [hunk 3]
```diff
@@ -533,5 +533,67 @@ var (
...
+			Stage:       FeatureStageExperimental,
+			Owner:       grafanaObservabilityMetricsSquad,
+		},
+		{
+			Name:         "alertingLokiRangeToInstant",
+			Description:  "Rewrites eligible loki range queries to instant queries",
+			Stage:        FeatureStageExperimental,
+			FrontendOnly: false,
+			Owner:        grafanaAlertingSquad,
+		},
```
Comment (JohnnyQQQQ) [inline pkg/services/featuremgmt/registry.go:597]:
```suggestion
		{
			Name:         "alertingLokiRangeToInstant",
			Description:  "Rewrites eligible loki range queries to instant queries",
			Stage:        FeatureStageExperimental,
			FrontendOnly: false,
			Owner:        grafanaAlertingSquad,
		},
```

File: pkg/services/featuremgmt/toggles_gen.go [hunk 4]
```diff
@@ -322,4 +322,40 @@ const (
...
+	// Improves the scrolling behavior of logs in Explore
+	FlagExploreScrollableLogsContainer = "exploreScrollableLogsContainer"
+
+	// FlagRecordedQueriesMulti
+	// Enables writing multiple items from a single query within Recorded Queries
+	FlagRecordedQueriesMulti = "recordedQueriesMulti"
+
+	// FlagAlertingLokiRangeToInstant
+	// Rewrites eligible loki range queries to instant queries
+	FlagAlertingLokiRangeToInstant = "alertingLokiRangeToInstant"
```
Comment (JohnnyQQQQ) [inline pkg/services/featuremgmt/toggles_gen.go:360]:
```suggestion

	// FlagAlertingLokiRangeToInstant
	// Rewrites eligible loki range queries to instant queries
	FlagAlertingLokiRangeToInstant = "alertingLokiRangeToInstant"
```

File: pkg/services/featuremgmt/registry.go [hunk 5]
```diff
@@ -533,5 +533,12 @@ var (
 			State:        FeatureStateBeta,
 			Owner:        grafanaBiSquad,
 		},
+		{
+			Name:         "alertingLokiRangeToInstant",
+			Description:  "Rewrites eligible loki range queries to instant queries",
+			State:        FeatureStateAlpha,
```
Comment (JohnnyQQQQ) [inline pkg/services/featuremgmt/registry.go:539]: This was called "stage" on main but is here its still called "state". It has not the value "experimental" available, so I used alpha.

**Review by grobinson-grafana — APPROVED** [review]

**Pull request merged**